MMR.Randomizer/Utils/SequenceUtils.cs

```diff
@@ -42,75 +42,99 @@ public static void ReadSequenceInfo()
+                    {
+                        Name = sourceName,
+                        Type = sourceType,
+                        Instrument = sourceInstrument
+                    };
 
-                SequenceInfo targetSequence = new SequenceInfo
-                {
-                    Name = targetName,
-                    Type = targetType,
-                    Instrument = targetInstrument
-                };
+                    SequenceInfo targetSequence = new SequenceInfo
+                    {
+                        Name = targetName,
+                        Type = targetType,
+                        Instrument = targetInstrument
+                    };
 
-                if (sourceSequence.Name.StartsWith("mm-"))
-                {
-                    targetSequence.Replaces = Convert.ToInt32(lines[i + 3], 16);
-                    sourceSequence.MM_seq = Convert.ToInt32(lines[i + 3], 16);
-                    RomData.TargetSequences.Add(targetSequence);
-                    if (lines[i+4] == "no-recycle"){
-                        Debug.WriteLine("Player does not want to reuse song: " + sourceSequence.Name);
-                        sourceSequence.Name = "drop";
-                        i += 1;
-                    }
-                    i += 4;
-                }
-                else
-                {
-                    i += 3;
-                    if (File.Exists(Path.Combine(Values.MusicDirectory, sourceName)) == false)
+                    if ( sourceSequence.Name.StartsWith("mm-") )
                     {
-                        // if sequence file doesn't exist, was removed by user, ignore it
-                        continue;
+                        targetSequence.Replaces = Convert.ToInt32(lines[i + 3], 16);
+                        sourceSequence.MM_seq = Convert.ToInt32(lines[i + 3], 16);
+                        if (lines[i+4] == "no-recycle"){
+                            //Debug.WriteLine("Player does not want to reuse song: " + sourceSequence.Name);
+                            sourceSequence.Name = "drop";
+                            i += 1;
+                        }
+                        i += 4;
+                        if (RomData.TargetSequences.Find(u => u.Name == sourceName) != null)
+                            continue; //old already have it
+                        RomData.TargetSequences.Add(targetSequence);
                     }
-                };
+                    else
+                    {
+                        i += 3;
+                        if (File.Exists(Path.Combine(directory, sourceName)) == false)
+                        {
+                            // if sequence file doesn't exist, was removed by user, ignore it
+                            continue;
+                        }
+                        try
+                        {
+                            using (var reader = new BinaryReader(File.Open(Path.Combine(directory, sourceName), FileMode.Open)))
```

Comment:
Why do we need to read the file and store it now?

Reply:
Perhaps we don't need to load them beforehand, although we already load all MMRS code before hand so that we don't need to read the zip files in two places, so the two are compatible. Also we don't have to worry about multi-folder searching for files if they are already loaded earlier.

Reply:
Changed it so that it loads the files last second as requested.